src/modules/projectSearchContext.js

```diff
@@ -30,6 +29,95 @@ class ProjectSearchContext {
             body: JSON.stringify(searchRequest)
         }).then(this.sessionManager.toJson);
     };
+
+    /**
+     * Search for traces in a project and map every trace to the callback.
+     * 
+     * @param {string|object} request The query as an HQL query string, or the full REST request for the /search
+     * @param {function} callback The callback that receives a trace result object
+     * @returns 
+     */
+    tracesMap = (request = '', callback) => {
+        const searchRequest = typeof request === 'string' ? {query: {human: request}} : request;
+        searchRequest.facets = []; // No facets allowed in streaming for now as the regex below doesn't understand them
+
+        const searchResultRegex = /^(\{("[a-z0-9]+"\:\s?("[a-z0-9]+"|[0-9]+|\[\]),?\s?)*"traces"\:\s?\[)/i
+
+        return this.sessionManager.gatekeeper(`/projects/${this.projectId}/traces/search`, {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(searchRequest)
+        }).then((response) => {
+            const reader = response.body.getReader();
+            const decoder = new TextDecoder();
+
+            let buffer = '';
+            let searchResult;
+
+            return reader.read().then(function processData({done, value}) {
+                if (done) {
+                    // The current buffer contains the closing json of the search result.
+                    // So let's complete this function with the json object without the trace objects
+                    return Promise.resolve(JSON.parse(searchResult + buffer));
+                }
+            
+                // value for fetch streams is an Uint8Array
+                buffer += decoder.decode(value); // Note: not every byte is a single character!
+
+                if (!searchResult) {
+                    // The root element is the trace search result
+                    const result = buffer.match(searchResultRegex);
+                    if (!result) {
+                        return reader.read().then(processData);
+                    }
+                    buffer = buffer.substring(result[0].length);
+                    searchResult = result[0];
+                }
+
+                if (buffer.length === 0) {
+                    return reader.read().then(processData);    
+                }
+
+                let start = 0;
+                let depth = 1;
+                let inEscape = false;
```

Comment:
maybe add a utility method for escaping data where you can pass your character you want to escape. Might also be a bit more easier to test that way.

(or of course NPM some third party escaper, the JavaScrpt way :P)

Reply:
Are there really no streaming JSON parsers for javascript?

Reply:
Well the result of the search is an object with an inner array, instead of a plain array. I haven't found any simple streaming parsers with that functionality...

Reply:
> maybe add a utility method for escaping data where you can pass your character you want to escape. Might also be a bit more easier to test that way.
> 
> (or of course NPM some third party escaper, the JavaScrpt way :P)

Extracted to separate method